design/ceph/ceph-nfs-ganesha.md

```diff
@@ -213,8 +225,115 @@ The following directories should **not** be shared between SSSD and other contai
+will be able to use the same service principal, which will be valuable for auto-scaling NFS servers
+in the future. The principal then becomes easy to construct from known CephNFS fields as
+`<principalName>/<namespace>-<name>@<realm>`.
+
+Additionally, `getaddrinfo()` doesn't return the hostname by default because the default
+`resolv.conf` in the pod does not check localhost. The pod's DNS config must be updated as shown.
+```yaml
+dnsConfig:
+  searches:
+    - localhost
+```
+
+#### Volumes
+Volumes that should be mounted into nfs-ganesha container to support Kerberos:
+1. `keytabFile` volume: use `subPath` on the mount to add the `krb5.keytab` file to `/etc/krb5.keytab`
+2. `configFiles` volume: mount (without `subPath`) to `/etc/krb5.conf.d/` to allow all files to
+   be mounted (e.g., if a ConfigMap has multiple data items or hostPath has multiple conf.d files)
+
+#### Ganesha config
+Should add configuration. Docs say Active_krb5 is default true if krb support is compiled in, but
+most examples have this explicitly set.
+Default PrincipalName is "nfs".
+Default for keytab path is reportedly empty. Rook can use `/etc/krb5.keytab`.
+
+Create a new RADOS object named `kerberos` to configure Kerberos.
+```ini
+NFS_KRB5
+{
+   PrincipalName = nfs ; # <-- set from spec.security.kerberos.principalName (or "nfs" if unset)
+   KeytabPath = /etc/krb5.keytab ;
+   Active_krb5 = YES ;
+}
+```
+
+Add the following line to to the config object (`conf-nfs.${nfs-name}`) to reference the new
+`kerberos` RADOS object. Remove this line from the config object if Kerberos is disabled.
+```
+%url "rados://.nfs/${nfs-name}/kerberos"
+```
+
+These steps can be done from the Rook operator.
+
+#### Kerberos config
+Rook should take steps to remove any default configuration. This means that it should create its own
+minimal krb.conf and ensure that any imported directories are empty. While it might be nice for some
+users to include the default configurations that are present in the container, it is extremely
+difficult in practice to adequately document the interactions between defaults that may change from
+time to time in the container image (or between different distros), and users will have to expend
+mental effort to understand how their configurations may override defaults. Upstream documentation
+for krb5.conf is unclear about file ordering and override behavior. Therefore, Rook will rely on
+users to specify nearly all of the configuration which will ensure users are able to easily supply
+the configuration they require.
+
+The minimal `/etc/krb5.conf` file Rook will create is as so:
+```ini
+includedir /etc/krb5.conf.d/ # include all user-defined config files
+
+[logging]
+default = STDERR # only log to stderr by default
+```
+
+#### Enabling kerberos for an export
+As of Ceph v17.2.3, the `ceph nfs ...` CLI tool is unable to create exports with Kerberos security
```

Comment:
Is there a known fix, or unknown at this point? How about linking to a ceph issue? Unless we can know the version where it's fixed, I'm not sure it's helpful to mention the currently version.
```suggestion
Currently the `ceph nfs ...` CLI tool is unable to create exports with Kerberos security
```

Reply:
When the fix will be available is unknown.